### DISCUSSION TITLE
After enable ipv6 in the docker, it works for only static ipv6 address, but not DHCP ipv6 address

### DISCUSSION THREAD

**Discussion by WallaceSUI** [moby/moby#45210] [Q&A]
### Description

Sorry to trouble all of you.
Recently, I follow the guidance to enable ipv6 for the docker in my linux machine

I directly edit the /etc/docker/daemon.json file for adding these parameters:
"ipv6": true,
"ip6tables": true,
"experimental": true,
"fixed-cidr-v6": "2001:db8:abc1::/64"
Then just restart the docker.

Then for example, here are two ipv6 addresses
static ipv6 address A
DHCP(dynamic) generated ipv6 address B

Case1:
When I use "ping6 A" in my linux machine, it works with response

Case2:
When I use "ping6 B" in my linux machine, it works but should have a zone identifier behind, like "B%eth0"

Case3:
when I use "docker exec -it myDocker bash" to get inside a docker, and then run "ping6 A", it works with response

Case4:
when I use "docker exec -it myDocker bash" to get inside a docker, and then run "ping6 B", it always throws the error message "ping6: sendto: Invalid argument"
Then I also try "ping6 B%eth0", it will be time out without any response.

For case4, I do not understand, for DHCP generated ipv6 address, why we cannot use it or ping6 it inside the docker? Why the docker cannot automatically assign a zone identifier to it? what is the problem here? how to fix it?

Thanks a lot. Really need your help.

### Reproduce

1: directly edit the /etc/docker/daemon.json file for adding these parameters:
"ipv6": true,
"ip6tables": true,
"experimental": true,
"fixed-cidr-v6": "2001:db8:abc1::/64"

2: systemctl reload docker

3: deploy a docker

4: prepare a DHCP enabled ipv6 and a static ipv6 address

5: do the case1-4 I mentioned in the description

### docker version

```bash
Docker version 20.10.14, build a224086
```

### docker info

```bash
Client:
 Context:    default
 Debug Mode: false

Server:
 Containers: 12
  Running: 12
  Paused: 0
  Stopped: 0
 Images: 27
 Server Version: 20.10.14
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runtime.v1.linux runc io.containerd.runc.v2
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 9cc61520f4cd876b86e77edfeb88fbcd536d1f9d
 runc version: v1.1.3-0-gf46b6ba
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 4.19.264-6.ph3
 Operating System: Photon OS/Linux
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 27.46GiB
 Name: sc2-pdp-ipv4-021-222.eng.vmware.com
 ID: RZV3:3ICM:ACWP:EY76:VM2Q:BDZR:PBMG:5B2F:CFII:GXVL:VM4C:BITE
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: true
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
 Default Address Pools:
   Base: 172.18.0.1/16, Size: 24
```

**Comment by sam-thibault** [comment]
I will move this to a discussion for now.  Maybe a bug will be created from the discussion.

**Comment by WallaceSUI** [comment]
Can I get a response for this? Thanks.
